dev/PushNotifications/PushNotificationUtility.h

```
@@ -123,4 +127,126 @@ namespace winrt::Microsoft::Windows::PushNotifications::Helpers
+
+            // Get the ServerId for the ComActivatorGuid
+            DWORD serverId = 0;
+            DWORD serverIdSize = sizeof(serverId);
+            THROW_IF_FAILED(RegGetValueW(
+                HKEY_LOCAL_MACHINE,
+                clsidPathWithKey.c_str(),
+                L"ServerId",
+                RRF_RT_REG_DWORD,
+                nullptr /* pdwType */,
+                &serverId,
+                &serverIdSize));
+
+            // Path: HKLM\SOFTWARE\Classes\PackagedCom\Package\{PackageFullName}\Server\{serverId}
+            std::wstring serverPath{ packagedRegistryPath + LR"(\Server\)" + std::to_wstring(serverId) };
+            WCHAR argumentsBuffer[MAX_VALUE_NAME];
+            DWORD argumentsBufferLength = sizeof(argumentsBuffer);
+            THROW_IF_FAILED(RegGetValueW(
+                HKEY_LOCAL_MACHINE,
+                serverPath.c_str(),
+                L"Arguments",
+                RRF_RT_REG_SZ,
+                nullptr /* pdwType */,
+                &argumentsBuffer,
+                &argumentsBufferLength));
+
+            std::wstring argumentString{ argumentsBuffer };
+
+            WCHAR exeBuffer[MAX_VALUE_NAME];
+            DWORD exeBufferLength = sizeof(exeBuffer);
+            THROW_IF_FAILED(RegGetValueW(
+                HKEY_LOCAL_MACHINE,
+                serverPath.c_str(),
+                L"Executable",
+                RRF_RT_REG_SZ,
+                nullptr /* pdwType */,
+                &exeBuffer,
+                &exeBufferLength));
+
+            std::wstring exeString{ exeBuffer };
+            THROW_IF_FAILED(GetExeNameFromPath(exeString));
+
+            wil::unique_cotaskmem_string exePath;
+            THROW_IF_FAILED(wil::GetModuleFileNameExW(GetCurrentProcess(), nullptr, exePath));
+
+            std::wstring exeToCheck{ exePath.get() };
+            THROW_IF_FAILED(GetExeNameFromPath(exeToCheck));
+
+            if (argumentString == argumentToCheck && exeString == exeToCheck)
```

Use THROW_HR_IF to filter out errors as a pattern instead of returning S_OK in a conditional prematurely.

As a general rule, return S_OK only at the end of a function call.

---

if (argumentString == argumentToCheck)
{
      THROW_HR_IF_MSG(E_FAIL, exeString != exeToCheck, "Caller RegistrationProcess is not the same as COM Server!");
      found = true;
      break;
}